ledger/src/blockstore.rs

```diff
@@ -3871,9 +3889,23 @@ pub fn create_new_ledger_from_name(
     genesis_config: &GenesisConfig,
     access_type: AccessType,
 ) -> (PathBuf, Hash) {
-    let ledger_path = get_ledger_path_from_name(name);
+    let (ledger_path, blockhash) =
+        create_new_ledger_from_name_auto_delete(name, genesis_config, access_type);
+    (ledger_path.into_path(), blockhash)
```

Comment:
Doing `ledger_path.into_path()` converts it from a TempDir to a regular PathBuf; this allows using the same function and providing an auto-delete and "regular" version